### ISSUE TITLE
Users without manage_options capability don't have a Dashboard/My home link in wp-admin navigation

### ISSUE THREAD

**Issue by BogdanUngureanu** [Automattic/jetpack#19506]
#### Steps to reproduce the issue

1. Make sure you login with a user role that doesn't have manage_options role (e.g. Author)
2. After login, you'll be redirected to the main page dashboard: `wp-admin`
3. Move to another page (e.g. posts)
4. You'll not be able to go back to the main dasboard since there's no link in the navigation for it.

#### What I expected

To have a menu item in navigation to should point the user to wp-admin.

**Screenshots**
Before nav unification, Author role users had a menu item `Dashboard` redirecting them to `wp-admin`
![Screenshot 2021-04-09 at 15 31 07](https://user-images.githubusercontent.com/995955/114407335-f4e30480-9bb0-11eb-82f0-0f296e0804b0.png)

After nav-unification, there's no longer a menu item for this:
![Screenshot 2021-04-09 at 15 32 38](https://user-images.githubusercontent.com/995955/114407358-fa404f00-9bb0-11eb-9641-0049bc242531.png)

**Comment by obenland** [comment]
Is this different from what was hotfixed in 666-gh-Automattic/wpcomsh and 667-gh-Automattic/wpcomsh?

If this is still an issue we should find a solution to it as soon as possible.

**Comment by BogdanUngureanu** [comment]
Yes, it's a different issue and it's caused because at the moment `My Home` menu item is shown only for users with `manage_options` capability. For example users with `Author` role will not have a menu `My Home` at all. 

Also, simply changing the capability from `manage_options` to `read` will not suffice because at the moment My Home points to `https://wordpress.com/home/site-name` and, from what I understand, these user roles should not have access to that page.

With this in mind, I'm not really sure if the menu item should be pointing to `/wp-admin/` or `https://wordpress.com/stats/day/site-name` (the default page for a user with the author role while on wordpress.com). The former would make sense being in WP-Admin while the latter while being in wordpress.com.

**Comment by jordesign** [comment]
With Nav Unification work back on the agenda again - I'm wondering about the importance of this one? We don't really lunk menu items to `/wp-admin/` anywhere else in the menu so I wonder if this the current behaviour is just ok now?

**Comment by obenland** [comment]
I think everyone should always have access to the dashboard. [The capability here](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L107) should probably be `read`?

**Comment by mmtr** [comment]
> [The capability here](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L107) should probably be `read`?

Unfortunately, as pointed above by @BogdanUngureanu, that's not enough, given that currently `wordpress.com/home/:site` is only accessible by administrators. 

If we change the capability of the "My Home" menu to `read`, we'd need to also change the content of `wordpress.com/home/:site` so it can be accessed by non-admins as well. I anticipate this would be a huge task in terms of effort.

Alternatively, we could update the "My Home" menu link to `wp-admin/index.php` for non-admins, while keeping the current link to `wordpress.com/home/:site` for administrators. This would be fairly simple, but I wonder if this new kind of disparity can cause some confusion.

**Comment by Copons** [comment]
It seems we could do a better job of handling the default view.

As it seems, users that don't have access to My Home would default to Stats.
This can be tested with a few simple steps:

- Prepare a user with its own site (where they are admin), and another site (even better if unlaunched) where they are author or contributor.
- Navigate to https://wordpress.com/, which will automatically open My Home of their main site. 👍 
- Now switch to the site where they are authors.
- They'll land on Stats.

## The Issue with Stats

This automatic redirection to Stats seems to happen for unlaunched sites as well.
These sites don't have a Stats item in menu, and their stats don't load (they just show infinite spinners).

<img width="1099" alt="Screenshot 2021-11-12 at 17 33 30" src="https://user-images.githubusercontent.com/2070010/141510195-2e2e51d0-7b4e-4ed3-88da-aba134e53eb0.png">

We might want to add the Stats menu item even if the site is unlaunched, and avoid trying to load stats by showing a message like "Stats are unavailable while the site is Coming Soon" instead.
Or, we might want to default to the Posts screen instead.

## The Issue with Inbox

So we are in the Stats page of an unlaunched site, with a low-capabilities user.
The sidebar doesn't have either the My Home or the Stats items, so now the first item is Inbox.

<img width="265" alt="Screenshot 2021-11-12 at 17 31 32" src="https://user-images.githubusercontent.com/2070010/141509763-2d98adb7-9a35-4ea9-885e-4cc2cfb3d6e2.png">

The Inbox page is [limited to `manage_options`](https://github.com/Automattic/wp-calypso/blob/4cbfaeaa41e59d2579ba27e6588495d1920a3b4a/client/my-sites/email/inbox/index.jsx#L72), although the menu item is added [with the `edit_posts` capability](https://github.com/Automattic/jetpack/blob/70b22abf74f999be83416ea66365d2a629c732eb/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L175).

In other words, the author user will be kind of encouraged to click on Inbox, given it's the first sidebar item, a position normally fulfilled by My Home, which sends them to...

<img width="659" alt="Screenshot 2021-11-12 at 17 31 24" src="https://user-images.githubusercontent.com/2070010/141509777-d60a4591-ef3b-441c-8707-81b0d526378a.png">

First and foremost, we should update the Inbox menu item with the correct capability.
Then, we would need to make sure that low-capabilities users would land on the appropriate screen, which I guess would be Stats or Posts (also depending on what we decided earlier).

cc @ianstewart

**Comment by Copons** [comment]
In https://github.com/Automattic/wp-calypso/issues/50945 a previous discussion regarding default screens.

**Comment by mmtr** [comment]
> Then, we would need to make sure that low-capabilities users would land on the appropriate screen, which I guess would be Stats or Posts (also depending on what we decided earlier).

We also have another alternative I proposed [above](https://github.com/Automattic/jetpack/issues/19506#issuecomment-964295967). We can keep My Home as the default screen for everyone. However, that screen would point to different URLs depending on the user capabilities:
- For admins: `wordpress.com/home/:site` (Calypso's My Home page).
- For non-admins: `wp-admin/index.php`(Core's WP Admin dashboard).

**Comment by Copons** [comment]
@mmtr Sounds reasonable to me.

Though, I also agree with the concern about confusion, but wonder how many users happen to be admin of a site and non-admin of another.

What if instead of calling both items "My Home" we went more Core-ish, and called them "My Home" and "Dashboard"? 🤔 
This would also kind of alleviate the fact that currently the only way to get to the wp-admin dashboard is from My Home, which is not available to non-admins.

**Comment by ianstewart** [comment]
+1 to @mmtr's proposal as a fallback.

What breaks if we allow users with the following [WordPress roles](https://wordpress.org/support/article/roles-and-capabilities/) to have access to My Home?

- Administrator
- Editor
- Author
- Contributor

**Comment by jordesign** [comment]
I would guess that there might be some checklist items that might not be able to be actioned by non admins?

But the remainder of the My Home content should be totally ok.

**Comment by Copons** [comment]
@Automattic/ajax this has been moved to our Iteration 3, requiring a bit of investigation work in advance.

### Tasks

For starters:

- [ ] Stopgap: @mmtr's [proposal](https://github.com/Automattic/jetpack/issues/19506#issuecomment-970187582)
  - [x] Show My Home menu item to admins (checked by default).
  - [ ] Show (wp-admin) Dashboard menu item to non-admins.

Then:

- [ ] Investigate how the various parts of My Home would behave with different user roles.
  - [ ] Launch checklist (likely only needed by admins).
  - [ ] Updates/upsell carousel (likely only needed by admins).
  - [ ] Quick links (do all links matter to all roles?).
  - [ ] Help Search (do all suggestions matter to all roles?).
  - [x] WP App links (checked by default, this works with any role).

Likely, the My Home will now show much less content to lower roles.

- [ ] Determine if it still makes sense to show My Home (instead of wp-admin Dashboard) to each roles.
  - [x] Admin (checked by default).
  - [ ] Editor.
  - [ ] Author.
  - [ ] Contributor.
- [ ] For the roles that are supposed to have access to My Home, update it to only show relevant content for those roles.
- [ ] Show the My Home menu item to roles allowed to access My Home.
- [ ] Show the (wp-admin) Dashboard menu item to roles not allowed to access My Home.
- [ ] (TBD) Add the Screen Options toggle in My Home and wp-admin Dashboard to switch back and forth.

**Comment by mmtr** [comment]
> (TBD) Add the Screen Options toggle in My Home and wp-admin Dashboard to switch back and forth.

This was discussed in https://github.com/Automattic/wp-calypso/issues/50945#issuecomment-869398348 and pd2C0z-3q-p2#comment-146, and we finally opted for not including the switcher in My Home. However, users can still access the WP Admin dashboard via the secondary menu in My Home:

<img width="1482" alt="Screen Shot 2021-07-05 at 11 50 12" src="https://user-images.githubusercontent.com/4452464/124453095-4c93a500-dd87-11eb-87ab-96e927f8b8af.png">

**Comment by mmtr** [comment]
> Stopgap: @mmtr's [proposal](https://github.com/Automattic/jetpack/issues/19506#issuecomment-970187582)

I’m having second thoughts now about this.

After analyzing the current behavior, I confirmed that the noted problem in the issue summary is no longer applicable. The issue says that after logging in, non-admins are redirected to `/wp-admin` and as soon as they navigate to a different page, they are unable to go back to `/wp-admin` using the menu.

However, on my tests (a non-a11n account who is an author on a single site), users are never redirected to `/wp-admin` after logging in, they land in the Stats page.


So I’m concerned now about introducing a new landing page for these users, who might prefer landing in the Stats page given that `/wp-admin` is IMO pretty much useless at this point (and they are not used to it). In other words, given that we no longer redirect to WP Admin, I’m afraid this stopgap may cause more problems than it solves.

At this point, I think it makes more sense to explore what’s needed to enable My Home for everyone (and to be honest, after a quick look, it should be more easily doable than it sounds).

**Comment by Copons** [comment]
I agree with this assessment @mmtr.
Showing My Home to all roles will basically trigger a domino effect that would sort out all awkward behaviours happening with the "default default screen".

- It's consistent across roles, and across sites.
- It gives a link to the wp-admin Dashboard to everyone (currently only available to admins despite the fact that all roles can access it with a direct URL).
- It contains infos specifically directed to _any_ WPCOM users (e.g. support pages or upsells), whereas wp-admin is more community oriented.

All in all, Calypso makes things very easy to adjust conditionally depending on the user capabilities, and even if non-admin would get a simple Stats + Quick Links + Help homepage, I'd still call it an improvement over sending them straight to Stats. 🤔 

---

As a follow up, we might want to double check how (if?) we determine a fallback screen on site switch.

As it seems, we tend to keep the user on the same screen when changing site.
This makes sense if they are on Stats or Posts, which are accessible by all roles.
But it gets confusing when we are on a screen that we cannot access in the next site.

Example 1:
- On your main site, enter the Jetpack screen (`/activity-log/:site`).
- Switch to a site where your user have a lower role (e.g. contributor).
- ⚠️ You'll get a "You are not authorized to view this page" screen.

Example 2:
- On your main site, enter the General Settings screen (`/settings/general/:site`).
- Switch to a site where your user have a lower role.
- ⚠️ You'll be redirected to the **all sites** Stats screen (`/stats/day`).

Ideally, with My Home accessible to all roles, we'll be able to also make these redirections finally consistent.

**Comment by mmtr** [comment]
My Home is now been enabled for non-admins on all WP.com sites, although Atomic users won't see the menu item until the Jetpack changes are deployed to the Atomic sites tomorrow 14th December (they will still land in My Home though – they simply won't see a menu item pointing to My Home).
